scripts/update_content_schema_artifacts.py

```diff
@@ -0,0 +1,234 @@
+                resolved_uris.append(value_["$ref"])
+    return resource_content, resolved_uris
+
+
+def create_resource(
+    content: dict, specification: Specification = DRAFT202012
+) -> Resource:
+    """
+    Create a resource object from content and specification.
+    """
+    return Resource.from_contents(contents=content, default_specification=specification)
+
+
+def update_registry(my_registry: Registry, new_resource: Resource) -> Registry:
+    """
+    Update a resource in the registry.
+
+    Args:
+        my_registry (Registry): The registry to be updated.
+        new_resource (Resource): The new resource to be updated.
+
+    Returns:
+        Registry: The updated registry with the new resource.
+    """
+    modified_registry = my_registry.remove(URI(new_resource.id()))
+    return modified_registry.with_resource(
+        uri=URI(new_resource.id()), resource=new_resource
+    )
+
+
+def export_registry(registry: Registry, export_dir: Path):
+    """
+    Export the contents of the registry resources to individual files.
+
+    Args:
+        registry (Registry): The registry containing the resources to be exported.
+        export_dir (Path): The directory where the exported files will be saved.
+    """
+    export_dir.mkdir(parents=True, exist_ok=True)
+    for uri, resource in registry.items():
+        file_path = export_dir / f"{uri.split('/')[-1]}.json"
+        with file_path.open("w", encoding="utf-8") as file:
+            json.dump(resource.contents, file, indent=4)
+
+
+def get_content_uris(
+    class_content_dir: Path = CLASS_CONTENT_DIR, relative_to: Path = ROOT
+) -> list[URI]:
+    """Return the list of URIs of the class content schemas. The URIs are relative to root path"""
+    return [
+        os.path.join(
+            os.path.relpath(class_content_dir, relative_to),
+            file.replace(".json", ""),
+        )
+        for file in os.listdir(class_content_dir)
+    ]
+
+
+def generate_artifacts(
+    content_schemas_registry: Registry, schema_uris: list
+) -> Registry:
+    """The routine to create a registry of the artifacts generated from content schemas."""
+
+    resolved_registry_uris = set()
+    # incorporate referenced sub-resources in the resource content
+    for schema_uri in schema_uris:
+        modified_content, resolved_uris = modify_content(
+            content_schemas_registry.contents(schema_uri),
+            content_schemas_registry.resolver(),
+        )
+        resolved_registry_uris.update(resolved_uris)
+        modified_resource = create_resource(modified_content)
+        updated_registry = update_registry(content_schemas_registry, modified_resource)
+    # remove the content of the resolved resources from the registry
+    for uri in resolved_registry_uris:
+        updated_registry = updated_registry.remove(uri)
+    return updated_registry
+
+
+def main(check: bool = False):
+    """The main routine."""
+    expected_registry = generate_artifacts(
```

Comment:
If I understood correctly, the content schema artifacts shall not contain any references to other schemas anymore but be flattened. Thus wouldn't it be sufficient to run the artifact generation and check if any other artifact files changed?